plugins/org.jboss.tools.playground.easymport.maven/src/org/jboss/tools/playground/easymport/maven/MavenProjectConfigurator.java

```diff
@@ -32,16 +35,72 @@
 import org.eclipse.jface.wizard.IWizard;
 import org.eclipse.m2e.core.MavenPlugin;
 import org.eclipse.m2e.core.internal.IMavenConstants;
+import org.eclipse.m2e.core.internal.lifecyclemapping.discovery.LifecycleMappingDiscoveryRequest;
 import org.eclipse.m2e.core.internal.project.ProjectConfigurationManager;
 import org.eclipse.m2e.core.project.IProjectConfigurationManager;
 import org.eclipse.m2e.core.project.LocalProjectScanner;
 import org.eclipse.m2e.core.project.MavenProjectInfo;
 import org.eclipse.m2e.core.project.MavenUpdateRequest;
 import org.eclipse.m2e.core.project.ResolverConfiguration;
+import org.eclipse.m2e.core.ui.internal.wizards.LifecycleMappingDiscoveryHelper;
+import org.eclipse.m2e.core.ui.internal.wizards.MappingDiscoveryJob;
+import org.eclipse.ui.internal.wizards.datatransfer.EasymportJob;
 import org.eclipse.ui.wizards.datatransfer.ProjectConfigurator;
 
 public class MavenProjectConfigurator implements ProjectConfigurator {
 	
+	private static class CumulativeMappingDiscoveryJob extends MappingDiscoveryJob {
+		private static CumulativeMappingDiscoveryJob INSTANCE;
+		private Set<IProject> toProcess;
+		private boolean started;
+		
+		public synchronized static CumulativeMappingDiscoveryJob getInstance() {
+			if (INSTANCE == null) {
+				INSTANCE = new CumulativeMappingDiscoveryJob();
+			}
+			return INSTANCE;
+		}
+		
+		private CumulativeMappingDiscoveryJob() {
+			super(null);
+			this.toProcess = Collections.synchronizedSet(new HashSet<IProject>());
+		}
+		
+		public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
+			try {
+				getJobManager().join(EasymportJob.class, monitor);
+			} catch (InterruptedException ex) {
+				throw new CoreException(new Status(IStatus.WARNING, Activator.getDefault().getBundle().getSymbolicName(), ex.getMessage(), ex));
+			}
+			synchronized (this.toProcess) {
+				this.started = true;
+			}
+			//Detect and resolve Lifecycle Mapping issues
+			LifecycleMappingDiscoveryRequest discoveryRequest = LifecycleMappingDiscoveryHelper.createLifecycleMappingDiscoveryRequest(toProcess, monitor);
+			if(discoveryRequest.isMappingComplete()) {
+				return Status.OK_STATUS;
+			}
+			//Some errors were detected
+			discoverProposals(discoveryRequest, monitor);
+			openProposalWizard(toProcess, discoveryRequest);
+			this.toProcess.clear();
```

Comment:
if an exception is thrown, toProcess won't be cleared, leaving you with leaked project references. Should clear in a global finally block